crates/vm/src/builtins/type.rs

```diff
@@ -32,7 +32,121 @@ use crate::{
 use indexmap::{IndexMap, map::Entry};
 use itertools::Itertools;
 use num_traits::ToPrimitive;
-use std::{borrow::Borrow, collections::HashSet, ops::Deref, pin::Pin, ptr::NonNull};
+use std::{
+    any::{Any, TypeId},
+    borrow::Borrow,
+    collections::HashSet,
+    marker::PhantomData,
+    ops::Deref,
+    pin::Pin,
+    ptr::NonNull,
+};
+
+/// Type-erased storage for extension module data attached to heap types.
+pub struct TypeDataSlot {
+    // PyObject_GetTypeData
+    type_id: TypeId,
+    data: Box<dyn Any + Send + Sync>,
```

Comment:
Hmmm, does it need to be `dyn`? I'd rather have it like:
```rust

struct TypeDataSlot<T> {
  ...
data: Box<T>,
}

impl<T> TypeDataSlot<T> 
where T: Any + Send + Sync {
   ...
}
```

Reply:
If TypeDataSlot<T> is a generic type, how does it be saved in HeapTypeExt?